src/users/users.service.ts

```diff
@@ -23,4 +23,11 @@ export class UsersService {
     const user = await this.em.findOne(User, { email });
     return user;
   }
+
+  async getDetails(userId: number): Promise<User> {
+    const user = await this.userRepository.findOneOrFail(userId, {
+      populate: ["student", "teacher"],
```

Comment:
[NIT] - Better to populate logically based on the role of the current user rather than both student and teacher.